### PR TITLE
docs: add developer notes about rootless docker setup

### PR DESCRIPTION
**Pull request by R-Rudolf**
Based on the earlier developer setup guide, after executing `make dev`, permission problems can occur in the web service, like the following:
```
immich_web               | failed to load config from /usr/src/app/vite.config.js
immich_web               | error when starting dev server:
immich_web               | Error: EACCES: permission denied, open '/usr/src/app/vite.config.js.timestamp-1727939078397-b40dceb29345c.mjs'
```

This is expected when using rootless docker, see https://github.com/moby/moby/issues/45919.
In this PR the developer guide is extended with notes to mitigate the problem, and comments are added to the dev composer file as well.

### CODE REVIEW

**Review by alextran1502 — APPROVED** [review]

**Comment by alextran1502** [comment]
Can you help with formatting the docs with `npm run format:fix`?

**Comment by R-Rudolf** [comment]
Ohh, sorry I mistakenly closed the PR and now I do not have the permission to re-open it.

For a minute I thought close means merge. I am used to other repository platform.

Pleae help re-open, and as I understand only a final review is missing from the maintainer.

**Comment by danieldietzler** [comment]
> Ohh, sorry I mistakenly closed the PR and now I do not have the permission to re-open it.
> 
> For a minute I thought close means merge. I am used to other repository platform.
> 
> Pleae help re-open, and as I understand only a final review is missing from the maintainer.

FYI you cannot just merge PRs lol

File: docker/docker-compose.dev.yml
Comment (danieldietzler) [inline docker/docker-compose.dev.yml:1]: This has some formatting changes, please revert them.

**Comment by zackpollard** [comment]
Hey @R-Rudolf, happy to get these changes merged in, but could you please remove the formatting changes mentioned above and make sure you're still happy with the changes?

**Pull request merged**